File: HW/HW10/hw10.py
import collections


class hw10:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        results = []

        queue = collections.deque()
        queue.append(root)

        while queue:
            queueLength = len(queue)
            level = []
            for i in range(queueLength):
               node = queue.popleft()
               if node:
                   level.append(node.val)
                   queue.append(node.left)
                   queue.append(node.right)
            if level:
                results.append(level)

        return results
  

File: HW/HW10/hw10.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import javax.swing.tree.TreeNode;

public class hw10{
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> results = new ArrayList<>();
        
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()){
            int queueLength = queue.size();
            List<Integer> level = new ArrayList<>();

            for (int i = 0; i < queueLength; i++){
                TreeNode node = queue.poll();
                if (node != null){
                    level.add(node.val);
                    queue.add(node.left);
                    queue.add(node.right);
                }
            }
              
            if (!level.isEmpty()){
                 results.add(level);
            }
        }
        return results;
    }
}
